src/libmw/test/tests/mmr/Test_Segment.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 The Litecoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <mw/mmr/MMR.h>
+#include <mw/mmr/MMRUtil.h>
+#include <mw/mmr/LeafSet.h>
+#include <mw/mmr/Segment.h>
+#include <boost/optional/optional_io.hpp>
+#include <unordered_set>
+
+#include <test_framework/TestMWEB.h>
+
+namespace std {
+ostream& operator<<(ostream& os, const mw::Hash& hash)
+{
+    os << hash.ToHex();
+    return os;
+}
+} // namespace std
+
+using namespace mmr;
+
+struct MMRWithLeafset {
+    IMMR::Ptr mmr;
+    ILeafSet::Ptr leafset;
+};
+
+static mmr::Leaf DeterministicLeaf(const size_t i)
```

Comment:
Does the `static` modifier actually do anything on these functions?

Reply:
Yes, `static` limits its scope to the current compilation unit (`Test_Segment.cpp`), and avoids polluting the global namespace.

`uint64_t` is probably better.  I'll use that instead.